ssl/statem/statem_clnt.c

```diff
@@ -234,8 +200,11 @@ int ossl_statem_client_read_transition(SSL *s, int mt)
      * Note that after a ClientHello we don't know what version we are going
      * to negotiate yet, so we don't take this branch until later
      */
-    if (s->method->version == TLS1_3_VERSION)
-        return ossl_statem_client13_read_transition(s, mt);
+    if (s->method->version == TLS1_3_VERSION) {
```

Comment:
Hmmm....well, yes it probably should, but there is some complexity here (the difference between s->version and s->method->version). That actually highlights a problem in the definition of SSL_IS_TLS13(), i.e. it returns true even in the case where we haven't *yet* negotiated TLSv1.3, but we might do. That's not right. I've added a new commit to fix that, which means that I have been able to fix the above to use the macro.